src/App.tsx

```diff
@@ -1,18 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
+const getVisibleItems = (perPage: number, currentPage: number) => {
+  const lastIndex = perPage * currentPage;
+
+  return items.slice(lastIndex - perPage, lastIndex);
+};
+
 export const App: React.FC = () => {
+  const [currPage, setCurrPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const [total] = useState(items.length);
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(event.target.value));
+    setCurrPage(1);
+  };
+
+  const onPageChange = (page: number) => {
+    setCurrPage(page);
+  };
+
+  const visibleItems = getVisibleItems(perPage, currPage);
+
+  const getFirstItem = () => {
+    return (currPage - 1) * perPage + 1;
+  };
+
+  const getLastItem = () => {
+    const last = currPage * perPage;
+
+    return last > items.length ? items.length : last;
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currPage} (items ${getFirstItem()} - ${getLastItem()} of ${total})`}
```

Comment:
Why not saving this values to constants instead of creating functions and calling it?